xGame/client/src/Components/MatchSelect/MatchSelect.js

```diff
@@ -1,14 +1,121 @@
+                                    variant="outlined"
+                                    style={{ width: "80%" }}
+                                    name="opponent"
+                                    label="Search for a user"
+                                    value={values?.["email"] || ""}
+                                    onChange={(e) => setFieldValue("opponent", e.target.value)}
+                                  />
+                                </Col>
+                              </Row>
+                              <br></br>
+                              <Row>
+                                <Col>
+                                  <div role="group" aria-labelledby="my-radio-group">
+                                      <p>
+                                      <div>Your starting color: {values.picked}</div>
+                                      <label>
+                                        <Field type="radio" name="picked" value="Black" />
+                                        Black
+                                      </label>
+                                      <br></br>
+                                      <label>
+                                        <Field type="radio" name="picked" value="White" />
+                                        White
+                                      </label>
+                                      </p>
+                                    </div>
+                                </Col>
+                              </Row>
+                              <Row className="pt-2">
+                                <Col className="d-flex justify-content-center">
+                                  <Button
+                                    color="primary"
+                                    variant="contained"
+                                    style={{ textTransform: "none" }}
+                                    onClick={formProps.handleSubmit}
+                                  >
+                                    Create Match
+                                  </Button>
+                                </Col>
+                                <Row className="pt-2">
+                                  <Col className="d-flex justify-content-center">
+                                    <Button
+                                      size="small"
+                                      color="primary"
+                                      style={{ textTransform: "none" }}
+                                      onClick={handleClose}
+                                    >
+                                      Cancel
+                                    </Button>
+                                  </Col>
+                                </Row>
+                              </Row>
+                            </Col>
+                          );
+                        }}
+                      </Formik>
+                </DialogContent>
+              </Dialog>
```

Comment:
Since this file is the MatchSelect component, all of the Dialog stuff should be extracted into its own component, something like "MatchCreationDialog" (which can also just go in the MatchSelect folder. Then you'd reference it from this file and just pass it the props it needs, like 
"<MatchCreationDialog` open={open} toggleOpen={toggleShowMatchCreation} />"